### PR TITLE
Format selection

### PR DESCRIPTION
**Pull request by terziele**
Adding a new `Format Selection` action to a mouse context menu. As discussed here https://github.com/zed-industries/zed/pull/16080 

Release Notes:

- Adding a `Format Selection` action to a mouse context menu.

### CODE REVIEW

File: crates/editor/src/editor.rs [hunk 1]
```diff
@@ -9966,13 +9966,38 @@ impl Editor {
...
+        &mut self,
+        _: &FormatSelection,
+        cx: &mut ViewContext<Self>,
+    ) -> Option<Task<Result<()>>> {
+        let project = match &self.project {
+            Some(project) => project.clone(),
+            None => return None,
+        };
+
+        let selections = self
```
Comment (terziele) [inline crates/editor/src/editor.rs:9982]: Is there an ideomatic way to get a multiselection without a cursor with a single char?

File: crates/editor/src/mouse_context_menu.rs [hunk 2]
```diff
@@ -169,6 +177,9 @@ pub fn deploy_context_menu(
                 .separator()
                 .action("Rename Symbol", Box::new(Rename))
                 .action("Format Buffer", Box::new(Format))
+                .when(!selections, |builder| {
```
Comment (terziele) [inline crates/editor/src/mouse_context_menu.rs:180]: Should the button be shown only when a LSP supports range formatting or something? Maybe something like a flag in settings a-la `formatSelectionEnabled: true`?

**Comment by ConradIrwin** [comment]
@terziele Sorry for the slow reply here! This is something that we want, but the code around language servers is very in flux as we rebuild remoting for SSH.

I am going to close this, but if you want to rebase onto latest main in a few days, that would be fantastic. @mikayla-maki and I are about to shred the formatting code again, so wait until that lands :D 

Otherwise if you'd like to pair on it: https://calendly.com/conradirwin/pairing

**Comment by terziele** [comment]
@ConradIrwin Hey, thanks for reply!
I've rebased my branch onto latest main. Should I open a new PR/reopen this? Or wait until the code is shredded? :)

**Comment by ConradIrwin** [comment]
@terziele we should be good to go now #18242 has landed, thanks for your patience!

**Comment by terziele** [comment]
@ConradIrwin Hey, just made a PR for formatting https://github.com/zed-industries/zed/pull/18752

**Pull request closed without merging**